### PR TITLE
Document gUnknown_03006298

### PR DESCRIPTION
**Pull request by kageru**
Just a simple `sed` replace I did while working on a romhack and wanted to contribute upstream.
`make [modern] compare` runs fine locally.
 **Discord contact info**
I’m on the pret server. kageru#7213.

### CODE REVIEW

File: src/battle_tower.c [hunk 1]
```diff
@@ -2224,11 +2224,11 @@ static void GetApprenticeMultiPartnerParty(u16 trainerId)
...
     }
 
-    gUnknown_03006298[0] = validSpecies[Random() % count];
+    gFactoryOpponentTeam[0] = validSpecies[Random() % count];
     do
     {
-        gUnknown_03006298[1] = validSpecies[Random() % count];
-    } while (gUnknown_03006298[0] == gUnknown_03006298[1]);
+        gFactoryOpponentTeam[1] = validSpecies[Random() % count];
+    } while (gFactoryOpponentTeam[0] == gFactoryOpponentTeam[1]);
```
Comment (GriffinRichards) [inline src/battle_tower.c:2231]: `FactoryOpponent` might be misleading, considering that it's used here for a non-opponent outside the factory

Comment (kageru) [reply src/battle_tower.c:2231]: Hm, true. It’s more like the team of an NPC in various battle facilities (just mainly factory). How would you call that? `gNpcFrontierTeam`?

Comment (kageru) [reply src/battle_tower.c]: I renamed it to `gFrontierNpcTeam`. Let me know if that’s better.

Comment (GriffinRichards) [reply src/battle_tower.c]: What about something like `gFrontierTempParty`? In any case we tend to use "party" over "team"

Comment (kageru) [reply src/battle_tower.c]: Sounds good to me.

**Pull request merged**